crates/autopilot/src/driver_api.rs

```diff
@@ -0,0 +1,45 @@
+use crate::driver_model::{execute, solve};
+use anyhow::{anyhow, Context, Result};
+use reqwest::Client;
+use shared::http_client::response_body_with_size_limit;
+use url::Url;
+
+const RESPONSE_SIZE_LIMIT: usize = 10_000_000;
```

Comment:
Not related to this PR, but wanted to bring it up here as this reminded me about it.

One thing that I think we should be mindful of is to start thinking in many places of deadlines in terms of blocks and not in terms of time. We already have a `current_block` stream in many places that we can use for something like `current_block.wait_for_next_blocks(10)` instead of `tokio::time::sleep()`.

I think in this particular case, time is the correct kind of deadline to use, but for things like waiting for transactions to mine, blocks are a better measurement (and more easily verifiable by untrusted parties).

Reply:
You are commenting on the size limit in bytes for responses of the driver API.  I think you wanted to comment on the deadline in the run loop.

I agree that in some places blocks might make sense and that it does *not* make sense for the /solve deadline.